File: 이건희/level2/연속 부분 수열 합의 개수.py
def solution(elements):
    element = 2 * elements
    result = set()
    num = 0
    
    for i in range(1, len(elements) + 1):
        for j in range(len(element) - i + 1):
            num = sum(element[j:j + i])
            result.add(num)
            
    return len(list(result))

File: 김찬우/프로그래머스 /Lv.2/연속부분수열합의개수.java
import java.util.*;
import java.util.stream.IntStream;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

class Solution {
    public int solution(int[] elements) {
        
       int[] list =  new int[elements.length * 2];
        
        HashSet<Integer> result = new HashSet<>();
        
        for(int i=0; i<elements.length; i++){
            
            list[i + elements.length] = elements[i];
            list[i] = elements[i];
                
        }
        
        for(int i = 1; i < elements.length; i++ ){
            
            for(int j = 0 ; j < elements.length; j++){
                int num = Arrays.stream(list,j, j+i).sum();
                result.add(num);
            }
            
        }
        
        
        return result.size()+1;
    }
}
